sonarts-core/src/tools/new-rule.ts

```diff
@@ -0,0 +1,271 @@
+  ruleMetadata["___RULE_NAME_DASH___"] = ruleNameDash;
+  ruleMetadata["___RULE_CLASS_NAME___"] = ruleClassName;
+  ruleMetadata["___RULE_TITLE___"] = ruleTitle;
+  ruleMetadata["___RSPEC_KEY___"] = rspecKey;
+
+  copyWithReplace(
+    `${templatesFolder}/rule.template_ts`,
+    `${rootFolder}/sonarts-core/src/rules/${ruleClassName}.ts`,
+    ruleMetadata,
+  );
+
+  const testPath = `${rootFolder}/sonarts-core/tests/rules/${ruleClassName}`;
+
+  fs.mkdirSync(testPath);
+
+  copyWithReplace(`${templatesFolder}/unitTest.template_ts`, `${testPath}/${ruleClassName}.test.ts`, ruleMetadata);
+}
+
+function updateSonarTsJson(rootFolder: string, ruleNameDash: string) {
+  let sonarTsJson = JSON.parse(fs.readFileSync(`${rootFolder}/sonarts-core/tslint-sonarts.json`).toString());
+  sonarTsJson.rules[ruleNameDash] = true;
+  fs.writeFileSync(
+    `${rootFolder}/sonarts-core/tslint-sonarts.json`,
+    stringify(sonarTsJson, { cmp: (a, b) => (a.key < b.key ? -1 : 1), space: 2 }) + "\n",
+  );
+}
+
+function getRuleTitleAndRspecKey(rspecRuleFolder: string, rspecId: string) {
+  try {
+    const fileText = fs.readFileSync(`${rspecRuleFolder}/${rspecId}.json`).toString();
+
+    let ruleData = JSON.parse(fileText);
+    return { ruleTitle: ruleData["title"], rspecKey: ruleData["ruleSpecification"] };
+  } catch (err) {
+    console.error("could not find metadata for the rule. Have you run the rspec-api (with correct parameters)?");
+    showHelp();
+    throw err;
+  }
+}
+
+function getDescription(rspecRuleFolder: string, rspecId: string) {
+  try {
+    let replaceDict: any = {};
+    replaceDict["<em>"] = "";
+    replaceDict["</em>"] = "";
+    replaceDict["<p>"] = "";
+    replaceDict["</p>"] = `\n`;
+    replaceDict["<code>"] = "`";
+    replaceDict["</code>"] = "`";
+    replaceDict["<pre>"] = "```typescript";
+    replaceDict["</pre>"] = "```";
+    replaceDict["<h2>"] = "## ";
+    replaceDict["</h2>"] = `\n`;
+
+    const fileText = fs.readFileSync(`${rspecRuleFolder}/${rspecId}.html`).toString();
+    return replace(fileText, replaceDict);
+  } catch (err) {
+    console.error("could not find metadata for the rule. Have you run the rspec-api?");
+    showHelp();
+    throw err;
+  }
+}
+
+function updateReadme(rootFolder: string, ruleTitle: string, ruleNameDash: string) {
+  const readmePath = `${rootFolder}/README.md`;
+  let readme = fs.readFileSync(readmePath).toString();
+
+  const lines = readme.split("\n");
+
+  let result = [];
+
+  let resultsBlock1 = [];
+  let resultsBlock2 = [];
+  let resultsBlock3 = [];
+
+  let ruleValue = `* ${ruleTitle} ([\`${ruleNameDash}\`])`;
+  resultsBlock1.push(ruleValue);
+  let ruleKey = `[\`${ruleNameDash}\`]: ./sonarts-core/docs/rules/${ruleNameDash}.md`;
+  resultsBlock2.push(ruleKey);
+
+  // 0: start
+  // 1: ## Rules
+  // 2: newlines
+  // 3: rules
+  // 4: newlines
+  // 5: rules 2
+  // 6: newline
+  // 7: after
+
+  let state = 0;
+
+  for (var i = 0; i < lines.length; i++) {

```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Expected a 'for-of' loop instead of a 'for' loop with this simple iteration ([typescript:S4138](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4138))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=michalb%2Fnew-rules-script&issues=AWC8Oukl_Q-0dz4JAzUS&open=AWC8Oukl_Q-0dz4JAzUS)

[ISSUE_KEY]: AWC8Oukl_Q-0dz4JAzUS